### PR TITLE
Implement parsing escaped pipe characters in outline parameters

### PR DESCRIPTION
**Pull request by Mark90**
Proposed implementation for https://github.com/pytest-dev/pytest-bdd/issues/333

Let me know what you think!

### CODE REVIEW

**Comment by Mark90** [comment]
I tried the 3.8 tox build locally before I made this PR, it reported the same black formatting error as travis just did. It's in `pytest_bdd/scripts.py` which hasn't been changed.
I recall having a similar issue for that file, when my local black changed `print ()` to `print()`, while CI wanted the former. Now it wants the latter. 
Not sure how to proceed :)

**Comment by Mark90** [comment]
I opted to just apply the black formatting to the offending file. Seems that the CI accepts the formatting now. However, both 3.5 and 3.6 testsuites report an error in `tests/feature/test_gherkin_terminal_reporter.py`


> E           AssertionError: assert '============================== 1 passed in 0.03s ===============================' == '============================== 1 passed in 0.02s ==============================='


This seems kind of arbitrary.

**Comment by The-Compiler** [comment]
It fails here:

https://github.com/pytest-dev/pytest-bdd/blob/e87b7563063c4c68e6daf1a9ee57de7f8e2a3995/tests/feature/test_gherkin_terminal_reporter.py#L175-L181

That code seems to account for that, but with pytest 5.1.0, the output format [changed](https://docs.pytest.org/en/latest/changelog.html#id76), breaking that normalizing...

**Comment by The-Compiler** [comment]
#340 should fix the test.

**Comment by Mark90** [comment]
@The-Compiler Thanks for the quick fix!

Will look into the change reported by codecov.

File: pytest_bdd/feature.py [hunk 1]
```diff
@@ -141,6 +142,18 @@ def get_tags(line):
     return set((tag.lstrip("@") for tag in line.strip().split(" @") if len(tag) > 1))
 
 
+def get_params(line):
```
Comment (youtux) [inline pytest_bdd/feature.py]: I think that "split_line" would be a better name for this, since we are not getting parameters from a line, but just splitting it with handling of escape sequences.

File: pytest_bdd/feature.py [hunk 2]
```diff
@@ -59,6 +59,7 @@
 
 STEP_PARAM_RE = re.compile(r"\<(.+?)\>")
 COMMENT_RE = re.compile(r"(^|(?<=\s))#")
+EXAMPLE_PARAM_RE = re.compile(r"(?<!\\)\|")
```
Comment (youtux) [inline pytest_bdd/feature.py]: I would call this `SPLIT_LINE_RE`

File: tests/feature/test_outline.py [hunk 3]
```diff
@@ -166,3 +166,71 @@ def test_outlined_feature(request):
         ["fruits"],
         [[u"oranges"], [u"apples"]],
     )
+
+
+def test_outline_with_escaped_pipes(testdir):
```
Comment (youtux) [inline tests/feature/test_outline.py]: The "when" and "then" steps are doing nothing, but we could improve the test by having "string1" being the string one under test, and string2 as the base64 representation of string1, and the "then" step can be something like "then <string2> should be the base64 encoding of <string1>.

The feature file would look like this (note the use of `r""" """` string, it allows to use less backslashes):
```
r"""\
    Feature: Outline With Special characters
        Scenario Outline: Outline with escaped pipe character
            Given we have strings <string1> and <string2>
            Then <string2> should be the base64 encoding of <string1>
            Examples:
            | string1        | string2          |
            | bork           | Ym9yaw==         |
            | \|bork         | fGJvcms=         |
            | bork \|        | Ym9yayB8         |
            | bork\|\|bork   | Ym9ya3x8Ym9yaw== |
            | \|             | fA==             |
"""
```

Let's also have these examples, where we check some edge cases with the escaping:
```
| bork \\\| | Ym9yayBcfA== |
| bork \\   | Ym9yayBc     |
```

Comment (Mark90) [reply tests/feature/test_outline.py]: Great idea! Just implemented and pushed it. The edge-case with 3 backslashes had a different base64 encoding actually, because the 3rd backslash is consumed by `split_line()`.

File: pytest_bdd/feature.py (hunk 1 above)
Comment (Mark90) [reply pytest_bdd/feature.py]: That makes a lot more sense, yes. Also removed the `if not line` check I had put in there, which I think was unnecessary.

File: pytest_bdd/feature.py (hunk 2 above)
Comment (Mark90) [reply pytest_bdd/feature.py]: Done

**Comment by youtux** [comment]
Thank you for the PR, @Mark90!
